src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -19,70 +19,99 @@ Status CoreWorkerPlasmaStoreProvider::Put(const RayObject &object,
 Status CoreWorkerPlasmaStoreProvider::Get(
     const std::vector<ObjectID> &ids, int64_t timeout_ms, const TaskID &task_id,
```

Comment:
In the future, we may want to assume that `ids` are unique and require the caller to deduplicate them. This wouldn't change the Ray API, but the Python code could dedup the IDs before passing them into C++ and then expand the results back out.

One potential reason for doing this is to "un-deduplicate" them after deserialization so that we reuse the same Python objects for the same IDs.

Basically, do we expect the following?

```python
x1, x2 = ray.get([x_id, x_id])

assert id(x1) == id(x2)  # Should this be true or false?
```

I think we probably want it to be true, though I'm not 100% sure.

Reply:
I wasn't sure what the desired behavior for this was, so decided it was safer to leave them as separate objects for now. My first instinct is that we probably want that condition to be `False`, because if the user wants a new reference to the same object they can easily accomplish it outside of Ray but if they want a new object backed by the same underlying data it's less clear how they would accomplish it.

I don't imagine that calling `ray.get()` on duplicate `ObjectIDs` is a common case anyways, but should definitely keep this on the radar.